src/Distribution/Nixpkgs/Haskell/FromCabal.hs

```diff
@@ -124,9 +125,16 @@ fromPackageDescription haskellResolver nixpkgsResolver missingDeps flags (Packag
     resolveInHackageThenNixpkgs i | haskellResolver (Dependency (mkPackageName (i^.ident)) anyVersion) = resolveInHackage i
                                   | otherwise = resolveInNixpkgs i
 
+    internalLibNames :: [PackageName]
+    internalLibNames = fmap unqualComponentNameToPackageName . catMaybes
+                     $ libName <$> subLibraries
+
+    internalLibDeps :: [Dependency]
+    internalLibDeps = targetBuildDepends . libBuildInfo =<< subLibraries
```

Comment:
How about the other dependencies recorded in a `BuildInfo` record? I assume that internal libraries can depend on system libraries and build tools, too?